clients/v1_auth/v1_auth.go

```diff
@@ -0,0 +1,423 @@
+package v1_auth
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/influxdata/influx-cli/v2/api"
+	"github.com/influxdata/influx-cli/v2/clients"
+	"github.com/influxdata/influx-cli/v2/pkg/influxid"
+)
+
+type Client struct {
+	clients.CLI
+	api.LegacyAuthorizationsApi
+	api.UsersApi
+	api.OrganizationsApi
+}
+
+type v1PrintOpts struct {
+	deleted bool
+	token   *v1Token
+	tokens  []v1Token
+}
+
+type v1Token struct {
+	ID          string   `json:"id"`
+	Description string   `json:"description"`
+	Token       string   `json:"token"`
+	Status      string   `json:"status"`
+	UserName    string   `json:"userName"`
+	UserID      string   `json:"userID"`
+	Permissions []string `json:"permissions"`
+}
+
+type CreateParams struct {
+	clients.OrgParams
+	Desc        string
+	Username    string
+	Password    string
+	NoPassword  bool
+	ReadBucket  []string
+	WriteBucket []string
+}
+
+func (c Client) Create(ctx context.Context, params *CreateParams) error {
+	if params.Password != "" && params.NoPassword {
+		return fmt.Errorf("only one of --password and --no-password may be specified")
+	}
+
+	orgID, err := c.getOrgID(ctx, params.OrgParams)
+	if err != nil {
+		return err
+	}
+
+	// verify an existing token with the same username doesn't already exist
+	auths, err := c.LegacyAuthorizationsApi.GetAuthorizations(ctx).Token(params.Username).Execute()
+	if apiError, ok := err.(api.ApiError); ok {
+		if apiError.ErrorCode() != api.ERRORCODE_NOT_FOUND {
+			return fmt.Errorf("failed to verify username %q has no auth: %w", params.Username, err)
+		} else if len(auths.GetAuthorizations()) != 0 {
+			return fmt.Errorf("authorization with username %q already exists", params.Username)
+		}
+	}
+	auth := auths.GetAuthorizations()[0]
+
+	password := params.Password
+	if password == "" && !params.NoPassword {
+		pass, err := c.StdIO.GetPassword("Please type your password")
+		if err != nil {
+			return err
+		}
+		password = pass
+	}
+
+	bucketPerms := []struct {
+		action string
+		perms  []string
+	}{
+		{action: "read", perms: params.ReadBucket},
+		{action: "write", perms: params.WriteBucket},
+	}
+
+	var permissions []api.Permission
+	for _, bp := range bucketPerms {
+		for _, p := range bp.perms {
+			// verify the input ID
+			if _, err := influxid.IDFromString(p); err != nil {
+				return fmt.Errorf("invalid bucket ID '%s': %w (did you pass a bucket name instead of an ID?)", p, err)
+			}
+
+			newPerm := api.Permission{
+				Action: bp.action,
+				Resource: api.PermissionResource{
+					Type:  "buckets",
+					Id:    &p,
+					OrgID: &orgID,
+				},
+			}
+			permissions = append(permissions, newPerm)
+		}
+	}
+	authReq := api.AuthorizationPostRequest{
+		Description: &params.Desc,
+		OrgID:       orgID,
+		Permissions: permissions,
+		UserID:      auth.UserID,
```

Comment:
This will need fixed up since we don't have `auth` (because of comment above).

It looks like the old CLI doesn't even set `UserID` (maybe it's derived from the auth token on the request?). You'll need to set `Token: params.Username` here